### PR TITLE
stm32h7: per comments, improved consistency with other rcc impls.

### PR DESCRIPTION
**Pull request by vielster**
Reduced the sea of enums, and renamed config parameters to match other
implementations, cribbing off of the G0 config, as it is closer to the level
of complexity. Updated initialization code to utilize the new values.

Added flash and LDO configuration from RCC init to be more consistent with
STM32 platform initialization.

### CODE REVIEW

File: lib/stm32/common/usart_common_all.c [hunk 1]
```diff
@@ -45,11 +45,15 @@ rcc_clock_setup-* functions in RCC).
...
 
+Note: This is not compatible with the complex clock tree implemented by the H7
+series of parts, and is defined separately for those modules.
+
 @param[in] usart unsigned 32 bit. USART block register address base @ref
 usart_reg_base
 @param[in] baud unsigned 32 bit. Baud rate specified in Hz.
 */
 
+#ifndef STM32H7
```
Comment (karlp) [inline lib/stm32/common/usart_common_all.c]: yeah, we're not goign to do this.  If there's a lot of other common code we do want, fine, then this goes somewhere else, but we're not goign to start putting ifndefs into this file.

**Comment by karlp** [comment]
The RCC stuff looks fine, but can you please separate out the uart stuff again?  we've got to find something else for that.

**Comment by vielster** [comment]
Split the usart stuff into a separate branch. I'll likely put up later...noodling with ideas...order of ideas in my head right now, at least in my preference...please lmk if you prefer any of these.

1. implement `rcc_get_peripheral_clk_freq(uint32_t periph_base)` for f/g/l devices, which presently only supports the USARTs, but can be extended to other peripherals. Note, this would be very useful for things like the SPI buses which have different clock bases per-each. I like this one the best as I think it gives the most flexibility, encourages adding to this implementation, gets away from globals, and is super simple to use in peripheral drivers. The global approach just can't really work for platforms with the clksel features, so this seems to be the most sensible (for now).

1. implement a unique `usart_setbaudrate()` for platform groups, or individual platforms (e.g. `g0/usart.c` OR `usart_common_g.c` and update makefiles appropriately. This works around the clock differences for now.

1. implement `rcc_get_peripheral_clk_freq(uint32_t periph_base)` as a weak symbol, provide global apb variables for h7, use the function if implemented, else use the old way. I don't like perpetuating the global feature in the h7, but this might be the least intrusive.

**Comment by karlp** [comment]
I'd say option 1 is probably best, we know we already have this on l4, f7, g0, and will on g4 as well, so it's kinda inevitable, I just want to make sure we get something ~sane early, and not chop too much into other code, or design it all based on h7 and then try and see how the rest of them need it.  I'd say 2 is your "easy" option to get a uart on h7 right now, and 3 is kinda yuk really, I agree with you.

**Comment by karlp** [comment]
merged in 4953d67a thanks!

**Pull request closed without merging**